packages/studio-ui/src/hooks/useInjectUserStyles.tsx

```diff
@@ -0,0 +1,74 @@
+import { useEffect } from "react";
+import useStudioStore from "../store/useStudioStore";
+import { USER_CUSTOM_STYLE_ATTRIBUTE } from "../utils/loadUserAssets";
+import isEqual from "lodash/isEqual";
+import { StudioStore } from "../store/models/StudioStore";
+
+/**
+ * This hook injects user styling from the Studio document head into the
+ * Studio Preview iframe's document head.  It only injects the relevant
+ * styling of the current active page and component tree.  Styling is cleared
+ * from the iframe before each injection.
+ */
+export default function useInjectUserStyles(
+  iframeDocument: Document | undefined
+) {
+  const userCssImports = useStudioStore(
+    (store) => getUserCssImports(store),
+    isEqual
```

Comment:
could you make a code maintenance item to investigate using `isEqual` as the default comparator for `useStudioStore`? I think that could be useful for reducing re-renders

Reply:
done! https://yexttest.atlassian.net/browse/SLAP-2974